drivers/hmis/spec/models/hmis/form/form_processor_spec.rb

```diff
@@ -813,14 +813,21 @@ def process_record(record:, hud_values:, user:, definition:, save: true)
     end
 
     it 'processing without saving does not persist unit assignment' do
-      # Assign existing enrollment to unit 1
+      # Set up existing enrollment that is already assigned to old_unit
       e2 = create(:hmis_hud_enrollment, data_source: ds1, project: p1, client: c1)
-      unit = create(:hmis_unit, project: p1)
-      expect(e2.current_unit).to be_nil
-      hud_values = complete_hud_values.merge('currentUnit' => unit.id)
-      process_record(record: e2, hud_values: hud_values, user: hmis_user, save: false, definition: definition)
-      e2.reload
-      expect(e2.current_unit).to be_nil
+      old_unit = create(:hmis_unit, project: p1)
+      e2.assign_unit(unit: old_unit, start_date: 1.week.ago, user: hmis_user)
+      e2.save!
+      e2.reload # reloading makes sure the enrollment's active_unit_occupancy is up-to-date
+      expect(e2.active_unit_occupancy.unit).to eq(old_unit)
+
+      # Use the form processor to reassign the enrollment to new_unit, but without saving
+      new_unit = create(:hmis_unit, project: p1)
+      hud_values = complete_hud_values.merge('currentUnit' => new_unit.id)
+      process_record(record: e2, hud_values: hud_values, user: hmis_user, definition: definition, save: false)
```

Comment:
```suggestion
      expect do
         process_record(record: e2, hud_values: hud_values, user: hmis_user, definition: definition, save: false)
     end.not_to change(Hmis::UnitOccupancy, :count)
```

Maybe worth adding this to ensure no occupancy is added? To prevent a silent failure where new_unit and old_unit are somehow both assigned, but line 830 still passes because old unit is considered active